vmdk_plugin/utils/fs/fs_windows.go

```diff
@@ -0,0 +1,242 @@
+func DevAttachWait(watcher *DeviceWatcher, volDev *VolumeDevSpec) error {
+	defer watcher.Terminate()
+	for {
+		log.WithFields(log.Fields{"volDev": *volDev}).Info("Waiting for a watcher event ")
+		select {
+		case event := <-watcher.Event:
+			log.WithFields(log.Fields{"volDev": *volDev,
+				"event": event}).Info("Watcher emitted an event ")
+			diskNum, err := getDiskNum(volDev)
+			if err != nil {
+				log.WithFields(log.Fields{"volDev": *volDev,
+					"err": err}).Warn("Couldn't map volDev to diskNum, continuing.. ")
+			} else {
+				log.WithFields(log.Fields{"volDev": *volDev,
+					"diskNum": diskNum}).Info("Successfully mapped volDev to diskNum ")
+				return nil
+			}
+			log.WithFields(log.Fields{"volDev": *volDev}).Warn("Couldn't locate disk, waiting.. ")
+
+		case err := <-watcher.Error:
+			log.WithFields(log.Fields{"volDev": *volDev,
+				"err": err}).Error("Watcher returned an error ")
+			return err
+
+		case <-time.After(maxDiskAttachWaitSec):
+			msg := "Disk mapping timed out "
+			log.WithFields(log.Fields{"volDev": *volDev}).Error(msg)
+			return errors.New(msg)
+		}
+	}
+}
+
+// DevAttachWaitFallback is a NOP.
+func DevAttachWaitFallback() {
+	// NOP since DevAttachWaitPrep never returns an error
+}
+
+// Mkdir creates a directory at the specified path.
+func Mkdir(path string) error {
+	return nil
+}
+
+// Mkfs creates a filesystem at the specified volDev.
+func Mkfs(fstype string, label string, volDev *VolumeDevSpec) error {
+	diskNum, err := getDiskNum(volDev)
+	if err != nil {
+		log.WithFields(log.Fields{"fstype": fstype, "label": label, "volDev": *volDev,
+			"err": err}).Error("Failed to locate disk ")
+		return err
+	}
+
+	// The diskpart utility consumes the script via stdin and formats a disk.
+	cmd := exec.Command(diskpart)
+	stdin, _ := cmd.StdinPipe()
+	defer stdin.Close()
+	io.WriteString(stdin, fmt.Sprintf(formatDiskScript, diskNum, ntfs, label))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		log.WithFields(log.Fields{"fstype": fstype, "label": label, "volDev": *volDev,
+			"diskNum": diskNum, "err": err}).Error("DiskPart script failed to execute ")
+		return err
+	} else if tailSegment(out, crlf, 5) != diskpartSuccessMsg {
+		log.WithFields(log.Fields{"fstype": fstype, "label": label, "volDev": *volDev,
+			"diskNum": diskNum, "out": string(out)}).Error("DiskPart failed to format the disk ")
+		return fmt.Errorf("DiskPart failed to format the disk for diskNum = %s", diskNum)
+	} else {
+		log.WithFields(log.Fields{"fstype": fstype, "label": label, "volDev": *volDev,
+			"diskNum": diskNum, "out": string(out)}).Info("DiskPart script executed successfully ")
+		return nil
+	}
+}
+
+// Mount mounts the filesystem on the volDev at the given mountpoint.
+func Mount(mountpoint string, fstype string, volDev *VolumeDevSpec, isReadOnly bool) error {
+	return nil
+}
+
+// Unmount unmounts a disk from the given mount point.
+func Unmount(mountpoint string) error {
+	return nil
```

Comment:
This isn't yet implemented. We'd be implementing this while working on volume attach.